client/middleware/complexRedirects.global.ts

```diff
@@ -0,0 +1,36 @@
+import 'core-js/actual/regexp/escape'
+import redirects from '../redirects.json'
+import { isMiddlewareRedirect } from '~/utilities/redirects'
+
+const middlewareRedirects = redirects.redirects
+  .filter((redirect) => isMiddlewareRedirect(redirect[0]))
+  .map((redirect): [RegExp, string] => {
+    // Convert wildcard '*' patterns into a regex for matching routes
+    // eg ["/rfc/rfc*.json", "/api/v1/rfc/rfc$1.json"]
+    // becomes [new RegExp('^/rfc/rfc(.*?)\.json$'), '/api/v1/rfc/rfc$1.json']
+
+    // @ts-expect-error RegExp.escape isn't natively supported by Node 22, hence the core-js polyfill above, so TS complains thinking it doesn't exist
+    const escapedPattern = RegExp.escape(redirect[0])
+
+    // Convert `*` to a regex pattern
+    const regexForPath = escapedPattern.replaceAll(
+      '\\*', // we need to match the escaped '*' as '\\*'
+      '(.*?)'
```

Comment:
Is the `?` doing anything here? `.*` matches 0 or more chars, doesn't it?

Reply:
It's a non-greedy expansion in case there were multiple wildcards